openpgp/packet/symmetrically_encrypted_aead.go

```diff
@@ -22,7 +22,7 @@ func (se *SymmetricallyEncrypted) parseAead(r io.Reader) error {
 
 	// Cipher
 	se.cipher = CipherFunction(headerData[0])
-	if se.cipher.KeySize() == 0 {
+	if !se.cipher.IsAes() {
```

Comment:
I don't think there's strictly speaking any requirement that AEAD needs to use AES, though we only support AES in OpenPGP.js too. But according to the crypto refresh it just needs to be a cipher with a block size of 16 bytes.

Reply:
I was undecided whether to call it `SupportsAead()` or `IsAes`. Since we do not support any other cipher for now this makes no functional difference. Shall I rename the function or make the function stricter (other comment)?

Reply:
Here I would just check `se.cipher.BlockSize() == 16`, since that's what the spec requires; I wouldn't make a dedicated function for that. Having a function `IsAES()` might still make sense for the cases where that's actually the requirement, such as for the KDF params (though if it's only there you could also just hardcode a check there instead of making a dedicated function).

Reply:
Mind if I make a function `SupportsAead()` that checks if the block size is 16, and `IsAes()` with the strengthened check and replace the usages?

Reply:
But then I fear we'd need a third `IsDeprecated()` for the MDC, and all these three functions would effectively do the same

Reply:
I don't think `SupportsAead()` makes sense, because it's not the cipher that does or doesn't support AEAD, it's the AEAD construction that does or doesn't support the cipher.

So tbh I would just hardcode a check in each place, in the way that most clearly checks exactly what we want to check, rather than making functions for each of them.

Reply:
Added a commit to fix this with explicit checks